src/index.ts

```diff
@@ -212,22 +221,49 @@ class TypeormJsonQuery<TEntity = ObjectLiteral> {
   /**
    * Get query attributes
    */
-  public getAttributes(attrs: IJsonQuery<TEntity>['attributes'] = []): string[] {
-    const { isDisableAttributes } = this.options;
-
-    const attributes = [
-      ...(isDisableAttributes ? [] : this.query.attributes ?? []),
-      ...attrs,
-      ...(this.authQuery.attributes || []),
-    ].map((field) => {
-      if (!field || typeof field !== 'string') {
-        throw new Error('Invalid json query: some attribute has incorrect name.');
-      }
-
-      return this.withFieldAlias(field);
-    });
+  public getAttributes(attrs: IJsonQuery<TEntity>['attributes'] = []): IJsonQueryAttribute[] {
```

Comment:
You can return: Rrquired<IJsonQueryAttribute>[]

Reply:
Here is another problem with type. I will add IAttribute internal interfaces, and if we will modify IJonQueryAttribute we will not have redundant fields in result of get attributes
<img width="813" alt="image" src="https://github.com/Lomray-Software/typeorm-json-query/assets/44732463/1e640b2c-3289-4cd1-a12b-b75f1786df50">